src/librustc_mir/transform/mod.rs

```diff
@@ -310,6 +353,8 @@ fn run_optimization_passes<'tcx>(
             &instcombine::InstCombine,
             &const_prop::ConstProp,
             &simplify_branches::SimplifyBranches::new("after-const-prop"),
+            // Deaggregation isn't technically an optimization but it unlocks opportunities
+            // for later optimization passes.
             &deaggregator::Deaggregator,
```

Comment:
But then why is it not run earlier?

Reply:
I don't know but I think the only passes that would be affected by this are `CopyPropagation` and `SimplifyLocals` which both run after this one.

Reply:
@eddyb said "people just probably didn't care much about it and added passes before and after".

So I'd vote for moving it up to the other lowering passes.

Reply:
I recall someone mentioning that drop elaboration relies on aggregates still being intact, and the generator transform performs drop elaboration. Not very sure though.